components/jira/actions/create-issue/create-issue.mjs

```diff
@@ -1,228 +1,216 @@
-      },
-    });
-
-    // Assumes the access token has access to a single instance
-    const cloudId = resp[0].id;
-
-    //Prepares the fields object of the request.
-    var fields = {
+    const fields = {
       summary: this.summary,
-      parent: {
-        key: this.parent_key,
-      },
       issuetype: {
-        id: this.issuetype,
+        id: this.issueType,
       },
       components: this.components,
       project: {
         id: this.projectID,
       },
-      description: this.description,
       reporter: {
-        id: this.reporter_id,
+        id: this.reporterId,
       },
-      fixVersions: this.fixVersions,
       labels: this.labels,
-      environment: this.environment,
-      versions: this.versions,
       duedate: this.duedate,
-      assignee: {
-        id: this.assignee_id,
-      },
+      ...this.customFields,
     };
-
-    if (this.priority_id) {
-      fields["priority"] = {
-        id: this.priority_id,
+    const description = this.jira.parseObject(this.description);
+    fields.description = description;
+    if (this.parentKey) {
+      fields.parent = {
+        key: this.parentKey,
       };
     }
-
-    if (this.originalEstimate || this.remainingEstimate) {
-      fields["timetracking"] = new Object();
-
-      if (this.remainingEstimate) {
-        fields["timetracking"]["remainingEstimate"] = this.remainingEstimate;
-      }
-
-      if (this.originalEstimate) {
-        fields["timetracking"]["originalEstimate"] = this.originalEstimate;
-      }
+    if (this.assigneeId) {
+      fields.assignee = {
+        id: this.assigneeId,
+      };
     }
-
-    if (this.security_id) {
-      fields["security"] = {
-        id: this.security_id,
+    if (this.priorityName) {
+      fields.priority = {
+        name: this.priorityName,
       };
     }
-
-    // Adds custom fields
-    if (this.custom_fields) {
-      var customFieldsArr = Object.keys(this.custom_fields);
-      for (var i = 0; i < customFieldsArr.length; i++) {
-        fields[customFieldsArr[i]] = this.custom_fields[customFieldsArr[i]];
-      }
+    if (this.remainingEstimate) {
+      fields.timetracking = {
+        remainingEstimate: this.remainingEstimate,
+      };
     }
-
-    // See https://developer.atlassian.com/cloud/jira/platform/rest/v3/#api-rest-api-3-issue-post
-    // for all options
-    return await axios($, {
-      method: "post",
-      url: `https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3/issue`,
-      headers: {
-        "Authorization": `Bearer ${this.jira.$auth.oauth_access_token}`,
-        "Accept": "application/json",
-        "Content-Type": "application/json",
-      },
+    if (this.originalEstimate) {
+      fields.timetracking = {
+        ...fields.timetracking,
+        originalEstimate: this.originalEstimate,
+      };
+    }
+    if (this.securityId) {
+      fields.security = {
+        id: this.securityId,
+      };
+    }
+    const update = this.jira.parseObject(this.update);
```

Comment:
Optional:
This function `parseObject` looks more like a utils function don't you think? How about moving it to something like `common/utils.mjs`?